tests/test_console.py

```diff
@@ -0,0 +1,461 @@
+            d = pickle.dumps({'foo': 'bar'})
+            print d
+            print pickle.loads(d)
+        """, """
+            (dp1
+            S'foo'
+            p2
+            S'bar'
+            p3
+            s.
+            {'foo': 'bar'}
+        """)
+        self.clear_console()
+        self.assertExecution("""
+            import pickle
+            print pickle.loads(pickle.dumps({'a': 'a'}))
+        """, """
+            {'a': 'a'}
+        """)
+
+    def test_js_module1(self):
+        self.assertExecution("""
+            import js
+            math = js.globals.Math
+            print(math.log(2))
+        """, """
+            0.6931471805599453
+        """)
+
+    def test_js_module2(self):
+        """
+        https://github.com/rfk/pypyjs/issues/56
+        """
+        self.assertExecution("""
+            import js
+            js.eval('var one = 123.456')
+            print js.eval('one')
+        """, """
+            123.456
+        """)
+
+    def test_module_random(self):
+        """
+        https://github.com/rfk/pypyjs/issues/4
+        """
+        self.assertExecution("""
+            import random
+            print random.__file__
+        """, """
+            /lib/pypyjs/lib_pypy/random.py
+        """)
+
+    def test_module_pprint(self):
+        """
+        https://github.com/rfk/pypyjs/issues/5
+        """
+        self.assertExecution("""
+            import pprint
+            print pprint.__file__
+        """, """
+            /lib/pypyjs/lib_pypy/pprint.py
+        """)
+
+    def test_module_platform(self):
+        """
+        https://github.com/rfk/pypyjs/issues/6
+        """
+        self.assertExecution("""
+            import platform
+            print platform.__file__
+        """, """
+            /lib/pypyjs/lib_pypy/platform.py
+        """)
+        
+    def test_module_os(self):
+        self.assertExecution("""
+            import os
+            print os.__file__
+        """, """
+            /lib/pypyjs/lib_pypy/os.py
+        """)
+
+    def test_imports(self):
+
+        # hack a list of available modules:
+        libpath = website_path("js/pypy.js-0.3.0/lib/modules")
+        module_names = [
+            os.path.splitext(item)[0]
+            for item in sorted(os.listdir(libpath))
+            if not item.startswith("_") and item.endswith(".py")
+        ]
+
+        good = failed = 0
+        total_count = len(module_names)
+        self.out("\nTry to import %i modules:" % total_count)
+        for no, module_name in enumerate(sorted(module_names)):
+            python_script = """
+                try:
+                    import %(name)s
+                except (ImportError, NameError) as err:
+                    print "Error importing '%(name)s': %%s" %% err
+                else:
+                    print "Import '%(name)s', ok."
+            """ % {"name":module_name}
+            self.clear_console()
+            self._execute_code(python_script)
```

Comment:
I think these should work if run using `exec()` rather than `_execute_code()`, since `exec` has some special logic to pull in the required `.py` files on demand.